### ISSUE TITLE
Fix chat_retrieval Streaming

### ISSUE THREAD

**Issue by krishokr** [PersonaFlow/agentstack#166]
## Issue
When streaming with an assistant of type `chat_retrieval`, we get the following error: 
``` 
{
    "status_code": 500,
    "message": "Internal Server Error: Expected dict, got [{'content': 'Hello.', 'additional_kwargs': {}, 'type': 'human', 'example': False}]"
}
```

https://github.com/user-attachments/assets/e4fd003d-55ad-4f7a-89fb-ba03ee3b01b2

**Comment by akaUNik** [comment]
Got the same error. Any fixes?

**Comment by danorlando** [comment]
> Got the same error. Any fixes?

Yes, there's an update for the checkpointer that needs to happen for the chat_retrieval architecture, which I've been meaning to get to so thanks for bringing it up... should have a fix up for it hopefully today.

**Comment by akaUNik** [comment]
@krishokr did you fixed this issue?
